app/models/service_provider.rb

```diff
@@ -158,15 +158,30 @@ def certs_are_pems
   end
 
   def validate_attribute_bundle
-    if (attribute_bundle || []).empty?
+    # attribute bundle should not be empty when saml and ial2 are selected
+    if ((attribute_bundle || []).empty? && ial == 2 && identity_protocol == 'saml')
```

Comment:
style nit, usually we don't add the extra parens if we don't need them
```suggestion
    if (attribute_bundle || []).empty? && ial == 2 && identity_protocol == 'saml'
```